src/main/java/net/neoforged/neoforge/capabilities/RegisterCapabilitiesEvent.java

```diff
@@ -0,0 +1,126 @@
+/*
+ * Copyright (c) NeoForged and contributors
+ * SPDX-License-Identifier: LGPL-2.1-only
+ */
+
+package net.neoforged.neoforge.capabilities;
+
+import java.util.ArrayList;
+import java.util.Objects;
+import net.minecraft.core.BlockPos;
+import net.minecraft.world.entity.Entity;
+import net.minecraft.world.entity.EntityType;
+import net.minecraft.world.item.Item;
+import net.minecraft.world.item.ItemStack;
+import net.minecraft.world.level.ItemLike;
+import net.minecraft.world.level.Level;
+import net.minecraft.world.level.block.Block;
+import net.minecraft.world.level.block.entity.BlockEntity;
+import net.minecraft.world.level.block.entity.BlockEntityType;
+import net.neoforged.bus.api.Event;
+import net.neoforged.fml.event.IModBusEvent;
+import org.jetbrains.annotations.ApiStatus;
+
+/**
+ * Fired to register capability providers at an appropriate time.
+ */
+public class RegisterCapabilitiesEvent extends Event implements IModBusEvent {
```

Comment:
We should provide some way for modded objects to also support capabilities and also register them with this event

Reply:
What I had in mind is that modded objects would have a corresponding event that would be fired from a `RegisterCapabilitiesEvent` handler. So nothing to be done on our end.

Reply:
having mods fire an event from a mod bus event seems like a bad idea, it would be nice if we could have some way to do it in here

Reply:
Why is this bad? A call to `ModLoader.get().postEvent(event)` will work just fine.

Reply:
@Technici4n unless it queues the event when all mods are loaded it could be bad since events could also be posted from the constructor of a mod when no listener is accessible.
And sometimes it isn't possible to delay that from the posters side.
(Like we had to write a complete custom config system to allow users to disable plugins for my mods because forges was loading to late and we had 0 control over it and it didn't support dynamic configs)

Reply:
Except that realistically it shouldn't happen until things are registered... so the simple solution would be to listen to this event in one place in your mod and then from there fire the other event for the custom ones? Why would you want to be firing a capability registration event before content exists? Without content there is nothing to actually attach said capabilities to.

Reply:
Can we consider this resolved?

Reply:
Marking as resolved - mods should fire an event at an appropriate time.